### PR TITLE
extract host from websocket to add route

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

Fixes: #8921

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 vmware-80u1

File: services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java [hunk 1]
```diff
@@ -284,6 +285,7 @@ private void connectClientToVNCServer(String tunnelUrl, String tunnelSession, St
         try {
             if (StringUtils.isNotBlank(websocketUrl)) {
                 s_logger.info(String.format("Connect to VNC over websocket URL: %s", websocketUrl));
+                ConsoleProxy.ensureRoute(NetUtils.extractHost(websocketUrl));
```
Comment (weizhouapache) [inline services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java:288]: @DaanHoogland 
`UriUtils` seems more suitable than `NetUtils`

Comment (DaanHoogland) [reply services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java:288]: I checked but UriUtils only deals with the class URI. I can move it though, I don't think it matters much.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test kvm-alma9 vmware-80

**Comment by DaanHoogland** [comment]
@blueorangutan test kvm-alma9 vmware-80u1

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 vmware-80u1

**Comment by weizhouapache** [comment]
test kvm-alma9 vmware-70u3

**Comment by weizhouapache** [comment]
@blueorangutan test alma9 vmware-70u3

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test matrix

**Review by weizhouapache — APPROVED** [review]
checked the vm console on vmware 8.0, looks good.

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
Verified VM consoles on VMware 6.5,7.0.3, KVM Rocky Linux 8.4, XCPng 8.2. No issues.

**Pull request merged**